docs/advanced/Channels.md

```diff
@@ -161,25 +161,38 @@ export function* saga() {
 }
 ```
 
-Here is another example of how you can use event channels to wait for events. Suppose you are waiting for an event that will trigger some event handler `onEvent`, which has yet to be defined. You'll want to define your `onEvent` function within your subscriber function:
+Here is another example of how you can use event channels to pass WebSocket events into your saga (using socket.io library). Suppose you are waiting for a server message 'ping' which you would like to consume with saga. You'll want to setup your subscription within your subscriber function:
 
 ```javascript
 import { take, put, call } from 'redux-saga/effects'
-import { eventChannel, END } from 'redux-saga'
+import { eventChannel, delay } from 'redux-saga'
+import socket from './socketConnection' // already established and exported connection
 
-export function* saga() {
+function* pong() {
+  yield call(delay, 5000)
+  socket.emit('pong')
+}
+
+export function* watchOnPings() {
+  const socketChannel = eventChannel(emit => {
```

Comment:
Ye, gonna amend it in few minutes
On Wed, 3 Aug 2016 at 13:02, Yassine Elouafi <notifications@github.com>
wrote:

> In docs/advanced/Channels.md
> <https://github.com/yelouafi/redux-saga/pull/464#discussion_r73314259>:
>
> >
> > -export function* saga() {
> > +function* pong() {
> > +  yield call(delay, 5000)
> > +  socket.emit('pong')
> > +}
> > +
> > +export function* watchOnPings() {
> > +  const socketChannel = eventChannel(emit => {
>
> It's not possible to test this form. Can we extract this into a separate a
> function like createWebSocketChannel and then call it using a declarative
> form yield call(createWebSocketChannel)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/yelouafi/redux-saga/pull/464/files/0c5927d0625bcce6e70238c70084592c9a9a8a65..2217327829bf6420bcf262ffd278acb3955a0f1e#r73314259>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJWMkmcXZtcJ-hTvoylReZfUC8jwSL7oks5qcG8WgaJpZM4JZkn9>
> .
>